src/App.jsx

```diff
@@ -6,7 +6,7 @@ import AddTodoForm from './AddTodoForm';
 
 function useSemiPersistentState() {
   const [todoList, setTodoList] = useState(
-    JSON.parse(localStorage.getItem('savedTodoList'))
+    JSON.parse(localStorage.getItem('savedTodoList')) || []
```

Comment:
The fallback to an empty array ensures the app doesn’t crash if localStorage is empty. Consider wrapping this logic in a try-catch block to handle potential parsing errors gracefully.